Source/Diagnostics/ComputeDiagFunctors/JFunctor.H

```diff
@@ -0,0 +1,47 @@
+#ifndef WARPX_JFUNCTOR_H_
+#define WARPX_JFUNCTOR_H_
+
+#include "ComputeDiagFunctor.H"
+
+#include <AMReX_BaseFwd.H>
+
+/**
+ * \brief Functor to cell-center MF for current density and store result in mf_out.
+ */
+class
+JFunctor final : public ComputeDiagFunctor
+{
+public:
+    /** Constructor.
+     *
+     * \param[in] mf_src source multifab.
+     * \param[in] lev level of multifab. Used for averaging in rz.
+     * \param[in] crse_ratio for interpolating field values from the simulation MultiFab, src_mf,
+                  to the output diagnostic MultiFab, mf_dst.
+     * \param[in] convertRZmodes2cartesian (in cylindrical) whether to
+     *            sum all modes in mf_src before cell-centering into dst multifab.
+     * \param[in] ncomp Number of component of mf_src to cell-center in dst multifab.
+     */
+    JFunctor(const int dir, const int lev,
```

Comment:
We tend to write `Function (...)` rather than `Function(...)` (extra white space between function's name and parenthesis) in function definitions, in order to make "git grepping" easier:
```suggestion
    JFunctor (const int dir, const int lev,
```